yjit/src/options.rs

```diff
@@ -29,6 +29,12 @@ pub struct Options {
 
     /// Verify context objects (debug mode only)
     pub verify_ctx: bool,
+
+    /// Whether or not to assume a global constant state (and therefore
+    /// invalidating code whenever any constant changes) versus assuming
+    /// constant name components (and therefore invalidating code whenever a
+    /// matching name component changes)
+    pub global_constant_state: bool,
```

Comment:
I feel iffy about having a command line option to control invalidation behavior. Invalidation is core to the JIT and needs to be tested well while adding a flag to control its behavior makes the testing and the review more difficult due to the added code path. It's not going to run on CI so it's also prone to code rot.

Reply:
That was my idea. I wanted to be able to get some stats with the old vs new invalidation behavior and it seemed more convenient. We could consider making this an environment variable instead (might be more convenient for Jean Boussier), and we can also plan to retire it once we've seen the data.